### ISSUE TITLE
Percolator query: Terms filter only regards first item in an object array

### ISSUE THREAD

**Issue by SuperMaggan** [elastic/elasticsearch#25101]
Version: 5.4.0

Percolator fields that contains a 'terms' filter that operate on an object array will only regard the first item in said array.
Let's set up the example:

### Create index
```
PUT /my-test
{
    "mappings": {
        "doctype": {
            "properties": {
                "object": {
                    "properties": {
                       "prop1": { 
                            "type": "long"
                        }
                    }
                }
            }        
        },
        "queries": {
            "properties": {
                "query": {
                    "type": "percolator"
                }
            }
        }
    }
}
```
### Insert saved search 
Now we're inserting an query item that matches all doctype items with object.prop1 = 1629
```
PUT /my-test/queries/1?refresh
{
    "query": {
      "bool": {
         "filter": [
            {
               "bool": {
                  "must": [
                     {
                        "terms": {
                           "object.prop1":[1629]                                       
                        }
                     }                                 
                  ]
               }
            }
         ]
      }
   }
}
```
### Percolate using doctype item
The following query should return our previous inserted saved query since it has an item object.prop1 = 1629. It will not though, since it's not the first item.
```
GET /my-test/_search
{
    "query" : {
        "percolate" : {
            "field" : "query",
            "document_type" : "doctype",
            "document" : {
                "message" : "kite4life",
                "object" : [
                    {
                        "prop1" : 16292
                    },
                    {
                        "prop1" : 1629
                    }                    
                ]
            }
        }
    }
}
```
However, if we flip the order, having object.prop1=1629 as the first item, it will return our previously saved search.

```
GET /my-test/_search
{
    "query" : {
        "percolate" : {
            "field" : "query",
            "document_type" : "doctype",
            "document" : {
                "message" : "kite4life",
                "object" : [
                    {
                        "prop1" : 1629
                    },
                    {
                        "prop1" : 16292
                    }                    
                ]
            }
        }
    }
}
```

Contact me if you need more examples/information.

//Magnus

**Comment by martijnvg** [comment]
@SuperMaggan Thanks for the clear example. This is indeed a bug. Somehow the prop1 field doesn't get indexed properly into the in-memory index.

**Comment by martijnvg** [comment]
The actual cause of this bug is in Lucene's `MemoryIndex`: https://issues.apache.org/jira/browse/LUCENE-7869

**Comment by SuperMaggan** [comment]
I'm impressed by your responsiveness! Thanks :)
Looking forward when I can drop my TermsToTerm() circumvention code.

**Comment by martijnvg** [comment]
This bug has been fixed in Lucene and the next 6.0 release will include that fix.
